app/controllers/api/v1/profiles_controller.rb

```diff
@@ -1,9 +1,26 @@
 class Api::V1::ProfilesController < Api::BaseController
+  before_action :authenticate_user, only: [:me]
+
   def show
     @user = User.find_by_slug!(params[:id])
     respond_to do |format|
       format.json { render json: @user }
       format.yaml { render yaml: @user }
     end
   end
+
+  def me
+    respond_to do |format|
+      format.json { render json: @user }
+      format.yaml { render yaml: @user }
+    end
+  end
+
+  private
+
+  def authenticate_user
+    authenticate_or_request_with_http_basic do |username, password|
```

Comment:
Looking at other usage of this auth method, we can make it similar (inline in action and keep the logic inside of this block).

https://github.com/rubygems/rubygems.org/blob/8f32c9f7e9f48e81926cf989af1a860a65a54a73/app/controllers/api/v1/api_keys_controller.rb#L4-L16